lib/Target/SubstraitPB/Import.cpp

```diff
@@ -47,13 +47,45 @@ namespace {
 DECLARE_IMPORT_FUNC(CrossRel, Rel, CrossOp)
 DECLARE_IMPORT_FUNC(FilterRel, Rel, FilterOp)
 DECLARE_IMPORT_FUNC(Expression, Expression, ExpressionOpInterface)
+DECLARE_IMPORT_FUNC(FieldReference, Expression::FieldReference,
+                    FieldReferenceOp)
 DECLARE_IMPORT_FUNC(Literal, Expression::Literal, LiteralOp)
-DECLARE_IMPORT_FUNC(NamedTable, Rel, NamedTableOp)
+DECLARE_IMPORT_FUNC(NamedTable, Rel, RelOpInterface)
 DECLARE_IMPORT_FUNC(Plan, Plan, PlanOp)
 DECLARE_IMPORT_FUNC(PlanRel, PlanRel, PlanRelOp)
 DECLARE_IMPORT_FUNC(ReadRel, Rel, RelOpInterface)
 DECLARE_IMPORT_FUNC(Rel, Rel, RelOpInterface)
 
+/// Imports the provided `RelCommon` message by producing an `EmitOp` that
+/// expresses the `Emit` message if it exists.
+///
+/// **This function must be called at the end of the import function of every
```

Comment:
Is it possible to have some top level helper function?

say

```c++

template <typename T>
mlir::FailureOr<RelOpInterface> importRel(..., T inputOp) {
  ... importRelImpl<T>(...)
  return importMaybeEmit(....)
}
```

Not sure if it helps that much, but makes a structure where more difficult to forget it.

Reply:
Yes, I think that's what I should aim for. I'll have a stab at that. The problem I'll have to solve for that is what I sketched above: I'll need to access the `common` field of the different message types but because they are different message types, there is no one function that does that. I think that the solution is to encapsulate that functionality in a function that does just that (and is implemented with one big switch).

Reply:
OK, that's what I ended up doing. I think that that was an important thing to clean up -- the new version is quite a bit more elegant.